service/storage_service.cc

```diff
@@ -5189,6 +5189,17 @@ future<raft_topology_cmd_result> storage_service::raft_topology_cmd_handler(shar
                 // we already did read barrier above
                 result.status = raft_topology_cmd_result::command_status::success;
             break;
+            case raft_topology_cmd::command::barrier_normal:
+                // we already did the barrier, but we need to check
+                // whether we are in normal mode
+                if (_operation_mode != mode::NORMAL) {
```

Comment:
`_operation_mode` is a legacy thing. We should use the state from the `topology` struct.